design/src/main/kotlin/uk/govuk/app/design/ui/component/Header.kt

```diff
@@ -40,27 +42,94 @@ fun TabHeader(
     }
 }
 
+data class ActionButton(
+    val text: String? = null,
+    val onClick: () -> Unit
+)
+
 @Composable
 fun ChildPageHeader(
-    text: String,
-    onBack: () -> Unit,
+    text: String? = null,
+    backButton: ActionButton? = null,
+    actionButton: ActionButton? = null,
     modifier: Modifier = Modifier,
 ) {
     Column(modifier) {
-        Box(
-            modifier = Modifier.height(64.dp),
-            contentAlignment = Alignment.CenterStart
-        ) {
-            TextButton(
-                onClick = onBack,
+        if (backButton != null || actionButton != null) {
+            Row(
+                modifier = Modifier
+                    .height(64.dp)
+                    .fillMaxWidth()
+                    .padding(GovUkTheme.spacing.small),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically,
             ) {
-                Icon(
-                    imageVector = Icons.AutoMirrored.Filled.ArrowBack,
-                    contentDescription = stringResource(R.string.content_desc_back),
-                    tint = GovUkTheme.colourScheme.textAndIcons.link
+                ChildPageHeaderBackButton(backButton, modifier = modifier.weight(1f))
+                ChildPageHeaderActionButton(
+                    actionButton,
+                    modifier = modifier.weight(1f)
                 )
             }
         }
+
+        if (text != null) {
+            ChildPageHeaderText(text = text, modifier = modifier)
+        }
+    }
+}
+
+@Composable
```

Comment:
I'm torn on these private functions. Having all of the child page header functionality inside the one function seems nicer in these design library files but I get it makes sense to break things down into smaller chunks. Happy to discuss.

Reply:
I felt  that the original version just had too much branching logic and `if` blocks - I guess I'm wired to break those out into functions. Happy to revert if that's the preference.

Reply:
Ok, again, it's a subjective thing so you can leave it in. Maybe at some point we will have to extract the components into their own files.

Reply:
Refactored - see https://github.com/alphagov/govuk-mobile-android-app/pull/142#discussion_r1876308662